core/src/trezor/wire/message_handler.py

```diff
@@ -0,0 +1,234 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from storage.cache_common import InvalidSessionError
+from trezor import log, loop, protobuf, utils, workflow
+from trezor.enums import FailureType
+from trezor.messages import Failure
+from trezor.wire.context import UnexpectedMessageException, with_context
+from trezor.wire.errors import ActionCancelled, DataError, Error, UnexpectedMessage
+from trezor.wire.protocol_common import Context, Message
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Container
+
+    from trezor.wire import Handler, LoadedMessageType, WireInterface
+
+    HandlerFinder = Callable[[Any, Any], Handler | None]
+    Filter = Callable[[int, Handler], Handler]
+
+# If set to False protobuf messages marked with "experimental_message" option are rejected.
+EXPERIMENTAL_ENABLED = False
+
+
+def wrap_protobuf_load(
+    buffer: bytes,
+    expected_type: type[LoadedMessageType],
+) -> LoadedMessageType:
+    try:
+        if __debug__:
```

Comment:
Do you still need this? Can we add `and utils.EMULATOR`? I'm a bit worried about performance of our hardware tests.

Reply:
Condition `and utils.EMULATOR` added here: https://github.com/trezor/trezor-firmware/pull/4357/commits/91d477c3c4a5ec331ce4410b11c941f94fe93cad

Reply:
Been thinking about this that it will probably fill CI logs with a lot of extra text, can we perhaps enable it when THP is enabled?

Reply:
Sure, why not: done here https://github.com/trezor/trezor-firmware/pull/4357/commits/d303dced7de86dead01563fa848532cb36be5811